libs/core/src/lib/carousel/carousel.component.ts

```diff
@@ -0,0 +1,412 @@
+    id = `fd-carousel-${carouselUniqueId++}`;
+
+    /** Sets aria-label attribute for carousel */
+    @Input()
+    ariaLabel = 'carousel';
+
+    /** Sets aria-labelledby attribute for carousel */
+    @Input()
+    ariaLabelledBy: string;
+
+    /** Sets aria-describedby attribute for carousel */
+    @Input()
+    ariaDescribedBy: string;
+
+    /** Sets position of page indicator container. Default position is bottom. */
+    @Input()
+    carouselIndicatorsOrientation: CarouselIndicatorsOrientation = 'bottom';
+
+    /** Height for carousel container */
+    @Input()
+    height: string;
+
+    /** Width for carousel container */
+    @Input()
+    width: string;
+
+    /** If carousel is in circular loop */
+    @Input()
+    loop = false;
+
+    /** Label for left navigation button */
+    @Input()
+    leftNavigationBtnLabel = 'Go to previous item';
+
+    /** Label for right navigation button */
+    @Input()
+    rightNavigationBtnLabel = 'Go to next item';
+
+    /** Shows/hides optional navigation button */
+    @Input()
+    navigation = true;
+
+    /** Show navigation button in page indicator container or inside content. Default is page indicator container on true value */
+    @Input()
+    navigatorInPageIndicator = true;
+
+    /** Convert to Numeric page indicator */
+    @Input()
+    numericIndicator = false;
+
+    /** Shows/hides optional page indicator container  */
+    @Input()
+    pageIndicatorContainer = true;
+
+    /** Shows/hides optional page indicator */
+    @Input()
+    pageIndicator = true;
+
+    /**
+     * An accessor that sets the resource strings.
+     * By default it uses EN resources.
+     */
+    @Input()
+    set resourceStrings(value: FdCarouselResourceStrings) {
+        this._resourceStrings = Object.assign({}, this._resourceStrings, value);
+    }
+
+    /**
+     * An accessor that returns the resource strings.
+     */
+    get resourceStrings(): FdCarouselResourceStrings {
+        return this._resourceStrings;
+    }
+
+    /**
+     * Returns the `role` attribute of the carousel.
+     */
+    @HostBinding('attr.role')
+    role = 'region';
+
+    /** Sets sliding duration */
+    @Input()
+    slideTransitionDuration = '150ms';
+
+    /** Is swipe enabled */
+    @Input()
+    swipeEnabled = true;
+
+    /**
+     * Returns the `tabIndex` of the carousel component.
+     */
+    @HostBinding('attr.tabindex')
+    get tabIndex(): number {
+        return 0;
+    }
+
+    /**
+     * Sets the overflow to auto value.
+     */
+    @HostBinding('style.overflow')
+    overflow = 'auto';
+
+    /** Is carousel is vertical. Default value is false. */
+    @Input()
+    vertical = false;
+
+    /** Number of items to be visible at a time */
+    @Input()
+    visibleSlidesCount = 1;
+
+    /** An event that is emitted after a slide transition has happened */
+    @Output()
+    readonly slideChange: EventEmitter<CarouselActiveSlides> = new EventEmitter<CarouselActiveSlides>();
```

Comment:
Could you put API properties in valid places?

Reply:
fixed